test/e2e/query_test.go

```diff
@@ -167,6 +170,53 @@ func TestQuery(t *testing.T) {
 	})
 }
 
+func TestQueryRoutePrefix(t *testing.T) {
+	t.Parallel()
+
+	s, err := e2e.NewScenario("e2e_test_query_route_prefix")
+	testutil.Ok(t, err)
+	defer s.Close()
+
+	q, err := e2ethanos.NewQuerier(
+		s.SharedDir(), "1",
+		nil,
+		nil,
+		nil,
+		"test",
+	)
+	testutil.Ok(t, err)
+	testutil.Ok(t, s.StartAndWaitReady(q))
+
+	ctx, cancel := chromedp.NewContext(context.Background())
+	defer cancel()
+
+	var networkErrors []string
+
+	// Listen for failed network requests and push them to an array.
+	chromedp.ListenTarget(ctx, func(ev interface{}) {
+		switch ev := ev.(type) {
+		case *network.EventLoadingFailed:
+			networkErrors = append(networkErrors, ev.ErrorText)
```

Comment:
Is chromedp guaranteeing to run this func sequentially? Otherwise we might have data races with concurrent access overwriting the networkErrors slice.

Reply:
Yes this is synchronous. You have to use goroutines if you want this to handle concurrently. They also recommend using goroutines almost all the time, but it was only an append statement so thought it would only add unnecessary complexity.

Reply:
ack good. the main thing I wanted to ensure was that we wouldn't have a data race. we aren't doing any blocking actions, we operating synchronously is fine :)